### PR TITLE
test(gnovm): improve gnovm/pkg/gnolang test coverage

### PR DESCRIPTION
**Pull request by mvertes**
No new tests have been added yet, but existing tests in gnovm/tests/files are now executed in the context of the package which contains the relevant interpreter virtual machine and parser code. We have now a better baseline to measure and complete the code coverage.

This represents more than 800 tests, which takes 10s on my macbook air m1. Those tests are actually run twice, as I haven't yet removed the original execution from gnovm/tests.

The testing code is substantially simplified compared to its counterpart in gnovm/tests, but has not yet been deduplicated. This will be done in further commits.

In `gnovm/pkg/golang`, `go test -cover` went from 34.2% to 66.7%.

Related to #1121, #1145 and probably more

<details><summary>Contributors' checklist...</summary>

- [*] Added new tests, or not needed, or not feasible
- [*] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [*] Updated the official documentation or not needed
- [*] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [*] Added references to related issues and PRs
- [*] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gnovm/pkg/gnolang/debugger_test.go [hunk 1]
```diff
@@ -23,49 +23,65 @@ type writeNopCloser struct{ io.Writer }
 
 func (writeNopCloser) Close() error { return nil }
 
-func eval(debugAddr, in, file string) (string, string, error) {
-	out := bytes.NewBufferString("")
-	err := bytes.NewBufferString("")
+// TODO (Marc): move evalTest to gnovm/tests package and remove code duplicates
```
Comment (thehowl) [inline gnovm/pkg/gnolang/debugger_test.go:26]: Note: I'm very happy if we want to keep `tests/` to contain test files only, while moving _how_ the tests are actually done into the relevant directories. To extend that idea, I kind of think that testscript (txtar) tests should go in `tests/`.

This is just to say, I think you can remove this error.

**Review by thehowl — APPROVED** [review]
:pray:

**Comment by thehowl** [comment]
branch got screwed up :thinking:

**Comment by mvertes** [comment]
I did a rebase instead of just merging master, maybe the cause?

**Comment by thehowl** [comment]
Yeah, I find that's generally the cause :wink:

**Review by moul — APPROVED** [review]

**Pull request merged**